thirdeye/thirdeye-frontend/app/pods/manage/alerts/edit/route.js

```diff
@@ -59,26 +63,30 @@ export default Ember.Route.extend({
         const currentStart = moment(currentEnd).subtract(1, 'months').valueOf();
         const baselineStart = moment(currentStart).subtract(1, 'week').valueOf();
         const baselineEnd = moment(currentEnd).subtract(1, 'week');
-        const url =  `/timeseries/compare/${metricId}/${currentStart}/${currentEnd}/` +
+        const metricDataUrl =  `/timeseries/compare/${metricId}/${currentStart}/${currentEnd}/` +
           `${baselineStart}/${baselineEnd}?dimension=${dimension}&granularity=${granularity}` +
           `&filters=${encodeURIComponent(formattedFilters)}`;
-        return fetch(url).then(checkStatus);
+        return fetch(metricDataUrl).then(checkStatus);
       })
       .then((metricData) => {
         Object.assign(metricData, { color: 'blue' })
         Object.assign(model, { metricData });
-
-        return fetch(`/thirdeye/email/functions`).then(checkStatus);
+        return fetch(`/thirdeye/entity/ALERT_CONFIG`).then(checkStatus);
       })
-      .then((groupConfigs) => {
-        // Temporary fix to match alert functions to subscribtion group
-        const subscriptionGroups = groupConfigs[id] || [];
-
-        // Back end supports 1-many relationships, however, we currently
-        // enforces 1-1 in the front end
-        const subscriptionGroup = subscriptionGroups.pop();
-
-        Object.assign(model, { subscriptionGroup });
+      .then((allConfigGroups) => {
+        // TODO: confirm dedupe
+        const uniqueGroups = _.uniq(allConfigGroups, name);
+        Object.assign(model, { allConfigGroups: uniqueGroups });
+        return fetch(`/thirdeye/email/function/${id}`).then(checkStatus);
+      })
+      .then((groupByAlertId) => {
+        const originalConfigGroup = groupByAlertId ? groupByAlertId.pop() : null;
+        const selectedAppName = originalConfigGroup ? originalConfigGroup.application : null;
+        Object.assign(model, { originalConfigGroup, selectedAppName });
+        return fetch('/thirdeye/entity/APPLICATION').then(checkStatus);
+      })
+      .then((allApps) => {
+        Object.assign(model, { allApps });
       });
```

Comment:
nit: do we need to add a `.catch` , can any of the calls above fail?

Reply:
Yes - we should add one, for sure.